### PR TITLE
Ubuntu 20.04 compatibility

### PR DESCRIPTION
**Pull request by ppenzin**
- Drive by cleanup: Build all wabt files in the build directory

### CODE REVIEW

**Comment by ppenzin** [comment]
This is not all, there are other errors coming from both palrt and the main sources. I think this is due to newer Clang being more pedantic about C++ type compatibility.

I am not sure why the definitions in `pal.h` in needed, but I need to get through the entire build to find out.

**Comment by rhuanjl** [comment]
These defines are wrapped in a `#ifndef PAL_STDCPP_COMPAT` could we try defining PAL_STDCPP_COMPAT and seeing if that fixes things?

**Comment by ppenzin** [comment]
We do define that, but not for PAL sources. I tried adding this to PAL, but things did not go well with definitions in `palinternal.h`. Using C99 standard does not help with this issue either, as it is not getting passed to C++ files.

I was able to get through a number of template parameter errors, but now stuck back in PAL on malloc/free/realloc, which PAL reimplements.

I am also bumping Ubuntu CI version, so that we can see the progress. Also, given the situation, maybe we should test with both 18.04 and 20.04.

**Review by ppenzin — COMMENTED** [review]
I tried to be consistent with these - the main goal is to simply give a different name to template parameters, so that they don't collide in situations like this:

```C++
template<class T>
class Foo {
  // Can't use the same name (T) here
  template<class T>
  friend class Bar;

  // ... and here
  template<class T>
  void Baz (T arg);
};
```

**Comment by ppenzin** [comment]
It was late at night and I changed Ubuntu image on copyright check job instead of builds. Also dropping PAL definitions change for now, will experiment and come up with something better.

**Comment by rhuanjl** [comment]
Interesting that Pal definitions are clashing with the Ubuntu stdlib. Could we just try deleting all the methods in pal that are in the stdlib?

**Comment by ppenzin** [comment]
If/when CI finishes it can be reviewed. I have added two more jobs for Ubuntu 18.04, not sure if they are needed, but thought it won't hurt since there are toolchain changes between those two versions.

_Edit:_ never mind, style check failed, as I forgot to update the copyright. Hold on tight.

**Comment by ppenzin** [comment]
Now as one big commit. There are mostly two things in it - template parameter disambiguation and PAL definition removal. It also adds two CI jobs for 18.04 and cleans up Wabt build set up.

File: azure-pipelines.yml [hunk 1]
```diff
@@ -33,18 +33,30 @@ jobs:
       maxParallel: 6
       matrix:
         Linux.Debug:
-          image_name: 'ubuntu-18.04'
+          image_name: 'ubuntu-20.04'
```
Comment (rhuanjl) [inline azure-pipelines.yml:36]: I agree we need the coverage - and let's take this in for now BUT I wonder if we could trim down slightly - maybe do debug on one platform and RelWithDebInfo on the other or something to reduce the CI run time.

**Review by rhuanjl — APPROVED** [review]

**Pull request merged**